syntax = "proto3";

service ChatService {
  rpc StartChat (StartChatRequest) returns (stream StartChatResponseChunk);
  rpc RequestChat (CompletionRequest) returns (stream CompletionResponseChunk);
}

// using empty is painful, and need for fields may arise
message StartChatRequest {}
message StartChatResponseChunk {
  string id = 1;
  int32 idx = 2;
  Role role = 3;
  string content = 4;
}

message CompletionRequest {
  string id = 1;
  string prompt = 2;
}

enum Role {
  UNDEFINED = 0;
  SYSTEM = 1;
  USER = 2;
  ASSISTANT = 3;
  TOOL = 4;
}

message CompletionResponseChunk {
  int32 idx = 1;
  Role role = 2;

  string content = 3;
}

